### PR TITLE
Add ability to configure matmul tilings from the Python API

### PR DESCRIPTION
**Pull request by Elarnon**
This allows manually overriding the tiling configuration for the MatMul
kernel through parameters in MatMulP and exposes that configuration
through the C and Python API.

In addition, a new Python example script is included which runs an
optimal candidate search from Python with a couple options (e.g. kernel
and device selection) which can be used as a starting point for further
experiments.

### CODE REVIEW

File: kernels/src/linalg.rs [hunk 1]
```diff
@@ -331,9 +337,21 @@ impl<'a, S: Scalar> Kernel<'a> for MatMul<'a, S> {
...
+        let m_tiles = if let Some(ref tiles) = self.params.m_tiling {
+            vec![tiles.clone()]
+        } else {
+            ::generate_tile_sizes(self.params.m as u32, &[64, 8])
+        };
+        let n_tiles = if let Some(ref tiles) = self.params.k_tiling {
+            vec![tiles.clone()]
+        } else {
+            ::generate_tile_sizes(self.params.n as u32, &[64, 8])
+        };
```
Comment (ulysseB) [inline kernels/src/linalg.rs]: I tend to prefer `unwrap_or_else` to `if let` when returning a value. But this is discutable.
```
let k_tiles = self.params.k_tiling.clone().map(Vec::new).unwrap_or_else(|| {
    ::generate_tile_sizes(self.params.k as u32, &[64])
})
```

File: telamon-capi/src/lib.rs [hunk 2]
```diff
@@ -15,6 +17,12 @@ use telamon::device::x86;
 use telamon::explorer::config::Config;
 pub use telamon_kernels::{linalg, Kernel};
 
+/// Initializes the logger.
+#[no_mangle]
+pub extern "C" fn env_logger_try_init() {
+    env_logger::try_init();
```
Comment (ulysseB) [inline telamon-capi/src/lib.rs]: You should use `let _ = env_logger::try_init();` instead to avoid warnings because of the unused `Result`.

Comment (Elarnon) [reply telamon-capi/src/lib.rs]: Ah, I was wondering what was the appropriate construct to use to silence those warnings! Will use this.

File: kernels/src/linalg.rs (hunk 1 above)
Comment (Elarnon) [reply kernels/src/linalg.rs]: That looks nicer; I'll change the code.

**Comment by ulysseB** [comment]
bors retry

**Comment by ulysseB** [comment]
My bad, I gave you the wrong code. Its `.map(|x| vec![x])` instead of `.map(Vec::new)`.

**Comment by Elarnon** [comment]
bors retry

**Pull request merged**